### ISSUE TITLE
Error when interacting with a Tradeshop

### ISSUE THREAD

**Issue by Darksoul676** [SparklingComet/TradeShop#38]
I'm currently running a 1.12 bukkit server with plugins and seem to have an error that posts to the console whenever a player interacts with a TradeShop sign, which prevents the sign itself from functioning correctly and doesn't post anything in-game. However, the error below posts to the console:

-------------------------------------------------------------------------------------------------------------------
```
[02:41:15] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to TradeShop v1.7.0-BETA
org.bukkit.event.EventException: null
at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-596221b-9a1fc1e]
at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-596221b-9a1fc1e]
at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [server.jar:git-Spigot-596221b-9a1fc1e]
at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [server.jar:git-Spigot-596221b-9a1fc1e]
at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:234) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.PlayerInteractManager.a(PlayerInteractManager.java:458) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:944) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-596221b-9a1fc1e]
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_151]
at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_151]
at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:747) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678) [server.jar:git-Spigot-596221b-9a1fc1e]
at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576) [server.jar:git-Spigot-596221b-9a1fc1e]
at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.lang.IllegalArgumentException: Provided data is not of type org.bukkit.material.MaterialData, found org.bukkit.material.TripwireHook
at org.bukkit.inventory.ItemStack.setData(ItemStack.java:251) ~[server.jar:git-Spigot-596221b-9a1fc1e]
at org.shanerx.tradeshop.Utils.itemCheck(Utils.java:268) ~[?:?]
at org.shanerx.tradeshop.Utils.canExchange(Utils.java:317) ~[?:?]
at org.shanerx.tradeshop.trade.TradeEventListener.onBlockInteract(TradeEventListener.java:148) ~[?:?]
at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source) ~[?:?]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_151]
at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_151]
at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-596221b-9a1fc1e]
... 17 more
```
-------------------------------------------------------------------------------------------------------------------

Originally, because I am running my server with multiple plugins, I had thought that the error may be generated as a result of a compatibility conflict. As I have no coding experience and had no idea what could be causing the issue, I tried running the plugin alone on a separate fresh-install server to test my theory but the problem persisted.

On an unrelated closing note I'd like to say that, despite this bug, I have thoroughly enjoyed the plugin so far. Thanks to all involved in taking the time to develop and manage it.

**Comment by SparklingComet** [comment]
Thank you for your report, we are happy to hear that you are enjoying the plugin even though you encountered this bug.
Currently, a few bugs were fixed but the latest .jar file has not been uploaded to SpigotMC yet. Would you be willing to test it again with the new jar as soon as I update it on the website? The error is unlikely to disappear, but we will at least see the up-to-date stacktrace.

**Comment by Darksoul676** [comment]
Sure thing! I'll be sure to test it as soon as possible and report any errors here if/when they occur.

**Comment by SparklingComet** [comment]
Excellent, thank you.
I will update the plugin’s page in a couple hours and notify you.

**Comment by KillerOfPie** [comment]
@Darksoul676 Did you do anything with custom items? or did the error pop up while you were making a shop?

**Comment by SparklingComet** [comment]
@Darksoul676 update’s out!

**Comment by Darksoul676** [comment]
@KillerOfPie I had been trying to test out custom items, yes, as I wanted to utilize the feature on my server in order to allow for a physical currency system that worked with admin/player shops. The error above posted to the console upon attempting to trade an emerald for the default 'Key' custom item.

**Comment by KillerOfPie** [comment]
Have you tried the new update yet? @Darksoul676

**Comment by Darksoul676** [comment]
@SparklingComet I have updated my server and tested the shops once more. This time the following errors occured:

-------------------------------------------------------------------------------------------------------------------

`[03:05:00] [Server thread/ERROR]: Could not pass event SignChangeEvent to TradeShop v1.7.2-BETA
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:2193) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PacketPlayInUpdateSign.a(SourceFile:44) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PacketPlayInUpdateSign.a(SourceFile:10) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-596221b-9a1fc1e]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_151]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_151]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:747) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576) [server.jar:git-Spigot-596221b-9a1fc1e]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.lang.NullPointerException
	at org.shanerx.tradeshop.itrade.IShopCreateEventListener.onSignChange(IShopCreateEventListener.java:66) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_151]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	... 15 more`

-------------------------------------------------------------------------------------------------------------------

`[03:05:03] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to TradeShop v1.7.2-BETA
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [server.jar:git-Spigot-596221b-9a1fc1e]
	at org.bukkit.craftbukkit.v1_12_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:234) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PlayerInteractManager.a(PlayerInteractManager.java:458) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PlayerConnection.a(PlayerConnection.java:944) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:37) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.PlayerConnectionUtils$1.run(SourceFile:13) [server.jar:git-Spigot-596221b-9a1fc1e]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_151]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_151]
	at net.minecraft.server.v1_12_R1.SystemUtils.a(SourceFile:46) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:747) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678) [server.jar:git-Spigot-596221b-9a1fc1e]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576) [server.jar:git-Spigot-596221b-9a1fc1e]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_151]
Caused by: java.lang.IllegalArgumentException: Provided data is not of type org.bukkit.material.MaterialData, found org.bukkit.material.TripwireHook
	at org.bukkit.inventory.ItemStack.setData(ItemStack.java:251) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	at org.shanerx.tradeshop.Utils.itemCheck(Utils.java:268) ~[?:?]
	at org.shanerx.tradeshop.Utils.canExchange(Utils.java:317) ~[?:?]
	at org.shanerx.tradeshop.itrade.ITradeEventListener.onBlockInteract(ITradeEventListener.java:134) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor14.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_151]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_151]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[server.jar:git-Spigot-596221b-9a1fc1e]
	... 17 more`

-------------------------------------------------------------------------------------------------------------------

**Comment by KillerOfPie** [comment]
@Darksoul676 would i be able to hop on the server and see what you're doing?

**Comment by Darksoul676** [comment]
@KillerOfPie Currently the server is being hosted locally, as my ISP isn't great. I would be more than happy to oblige if we could figure something out though.

**Comment by KillerOfPie** [comment]
@Darksoul676 Do you have discord?

**Comment by Darksoul676** [comment]
@KillerOfPie I do, yes. **********

*Edited by KOP to remove his discord name

**Comment by KillerOfPie** [comment]
Fixed with PR #40
